### PR TITLE
[SPARK-44266][SQL] Move Util.truncatedString to sql/api

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Move Util.truncatedString to sql/api. 

### Why are the changes needed?

Make StructType depends less on Catalyst so towards simpler DataType interface.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing test

### CODE REVIEW

**Comment by amaliujia** [comment]
R @cloud-fan 
cc @hvanhovell

File: sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala [hunk 1]
```diff
@@ -63,3 +66,39 @@ class StringConcat(val maxLength: Int = ByteArrayUtils.MAX_ROUNDED_ARRAY_LENGTH)
...
+    seq: Seq[T],
+    start: String,
+    sep: String,
+    end: String,
+    maxFields: Int): String = {
+    if (seq.length > maxFields) {
+      if (truncationWarningPrinted.compareAndSet(false, true)) {
+        logWarning(
+          "Truncated the string representation of a plan since it was too large. This " +
+            s"behavior can be adjusted by setting 'spark.sql.debug.maxToStringFields'.")
```
Comment (amaliujia) [inline sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala:90]: This is an example of that the error message actually only needs to access the SQLConf key `${SQLConf.MAX_TO_STRING_FIELDS.key}` so I replaced that with the key itself to eliminate the need of SQLConf. 

@cloud-fan please let me know if this is the proper way that you were thinking of.

File: sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala [hunk 2]
```diff
@@ -63,3 +66,39 @@ class StringConcat(val maxLength: Int = ByteArrayUtils.MAX_ROUNDED_ARRAY_LENGTH)
...
+  private val truncationWarningPrinted = new AtomicBoolean(false)
+
+  /**
+   * Format a sequence with semantics similar to calling .mkString(). Any elements beyond
+   * maxNumToStringFields will be dropped and replaced by a "... N more fields" placeholder.
+   *
+   * @return the trimmed and formatted string.
+   */
+  def truncatedString[T](
+    seq: Seq[T],
```
Comment (cloud-fan) [inline sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala:81]: nit: 4 spaces indentation

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**